crypto/multi_signature.go

```diff
@@ -0,0 +1,89 @@
+package crypto
+
+import (
+	"reflect"
+
+	"github.com/relab/hotstuff"
+	"golang.org/x/exp/slices"
+)
+
+// Signature is the individual component in MultiSignature
+type Signature interface {
+	Signer() hotstuff.ID
+	ToBytes() []byte
+}
+
+// MultiSignature is a set of (partial) signatures.
+type MultiSignature[T Signature] map[hotstuff.ID]T
+
+// RestoreMultiSignature should only be used to restore an existing threshold signature from a set of signatures.
+func RestoreMultiSignature[T Signature](signatures []T) MultiSignature[T] {
```

Comment:
Should rename this too: `RestoreMultiSignature` to `RestoreMulti` or just `Restore`